packages/file-graph/fileGraph.test.js

```diff
@@ -0,0 +1,110 @@
+import FileGraph from './fileGraph.js'
+import FileNode from './fileNode.js'
+
+describe('FileGraph', () => {
+  let root
+
+  beforeEach(() => {
+    root = new FileNode({
+      path: '/',
+      type: 'folder',
+      discovered: true,
+    })
+  })
+
+  test('add files to graph', () => {
+    const fileGraph = new FileGraph(root)
+
+    const filesToAdd = [
+      {
+        path: '/home/user/file1.txt',
+        type: 'file',
+        discovered: true,
+      },
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+        discovered: true,
+      },
+    ]
+
+    fileGraph.add(filesToAdd)
+
+    console.log(fileGraph.root.toString())
+    expect(fileGraph.root.children.length).toBe(1)
+    expect(fileGraph.root.children[0].children[0].children.length).toBe(2)
+  })
+
+  test('remove files from graph', () => {
+    const fileGraph = new FileGraph(root)
+
+    const filesToAdd = [
+      {
+        path: '/home/user/file1.txt',
+        type: 'file',
+        discovered: true,
+      },
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+        discovered: true,
+      },
+    ]
+
+    fileGraph.add(filesToAdd)
+
+    const filesToRemove = [
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+      },
+    ]
+
+    fileGraph.remove(filesToRemove)
+
+    console.log(fileGraph.root.toString())
+    expect(fileGraph.root.children[0].children[0].children.length).toBe(1)
+  })
+
+  test('discover new files', () => {
+    const fileGraph = new FileGraph(root)
+
+    const filesToAdd = [
+      {
+        path: '/home/user/file1.txt',
+        type: 'file',
+        discovered: true,
+      },
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+        discovered: true,
+      },
+    ]
+
+    fileGraph.add(filesToAdd)
+    expect(fileGraph.root.children[0].children[0].children.length).toBe(2)
+
+    const newFiles = [
+      {
+        path: '/home/newfile.txt',
+        type: 'file',
+        discovered: true,
+      },
+      {
+        path: '/home/user/file1.txt',
+        type: 'file',
+        discovered: true,
+      },
+    ]
```

Comment:
This is not a completed case for discover, please create more test cases to include:
1. merge children (what you've done)
2. merge in parent (which should keep the children those parents are not changed)